### PR TITLE
Folding report page sections.

### PR DESCRIPTION
**Pull request by isoos**
#7535

### CODE REVIEW

File: app/lib/frontend/templates/report.dart [hunk 1]
```diff
@@ -2,23 +2,34 @@
...
-  ModerationSubject? subject,
+  required ModerationSubject subject,
 }) {
+  final kindLabel = _subjectKindLabels[subject.kind] ?? 'about';
+  // TODO: also add `url`
+  final lcpsDeepLink =
+      Uri.parse('https://reportcontent.google.com/troubleshooter').replace(
+    queryParameters: {
+      'product': 'dart_pub',
+      'content_id': subject.fqn,
```
Comment (jonasfj) [inline app/lib/frontend/templates/report.dart:30]: `content_id` should be a canonical URL for the content in question.

One of:

* `pub.dev/publishers/<publisher>`
* `pub.dev/packages/<package>`
* `pub.dev/packages/<package>/versions/<version>`

```suggestion
      'content_id': subject.fqn, // TODO: Use subject.canonicalUrl, also add canonicalUrl
```

File: app/lib/admin/models.dart [hunk 2]
```diff
@@ -189,6 +189,8 @@ class ModerationSubject {
   }
 
   late final fqn = '$kind:$localName';
+  bool get hasPackage => package != null;
+  bool get hasPublisher => publisherId != null;
```
Comment (jonasfj) [inline app/lib/admin/models.dart:193]:
```suggestion
  bool get isPublisher => publisherId != null;
```

Comment (jonasfj) [reply app/lib/admin/models.dart:193]: Less, sure how `hasPackage` should be named :D

**Review by jonasfj — APPROVED** [review]

File: app/lib/admin/models.dart (hunk 2 above)
Comment (isoos) [reply app/lib/admin/models.dart]: Renamed to `isPackage` to keep it consistent.

**Pull request merged**